lib/rules/comma-dangle.js

```diff
@@ -13,14 +13,17 @@ const lodash = require("lodash");
 
 /**
  * Checks whether or not a trailing comma is allowed in a given node.
- * `ArrayPattern` which has `RestElement` disallows it.
+ * `ArrayPattern` which has `RestElement` or `ObjectPattern` which has `RestProperty` disallows it.
  *
  * @param {ASTNode} node - A node to check.
  * @param {ASTNode} lastItem - The node of the last element in the given node.
  * @returns {boolean} `true` if a trailing comma is allowed.
  */
 function isTrailingCommaAllowed(node, lastItem) {
-    return node.type !== "ArrayPattern" || lastItem.type !== "RestElement";
+    return !(
+        (node.type === "ArrayPattern" && lastItem.type === "RestElement") ||
+        (node.type === "ObjectPattern" && (lastItem.type === "RestProperty" || lastItem.type === "ExperimentalRestProperty"))
```

Comment:
Incidentally, why does espree add the `Experimental` prefix? I can understand the need to mark it as experimental, but I would think that is already accomplished by naming the option `experimentalObjectRestSpread`. When the node name eventually does change, it will almost certainly break downstream users, and that seems unnecessary to me. (Each node can only have one type, so this isn't like CSS vendor prefixes where the experimental and final versions can be supported simultaneously.)